core/templates/dev/head/pages/exploration_editor/EditorServices.js

```diff
@@ -550,9 +550,16 @@ oppia.factory('explorationRightsService', [
         var that = this;
 
         requestParams.version = explorationData.data.version;
-        var explorationRightsUrl = (
-          '/createhandler/rights/' + explorationData.explorationId);
-        $http.put(explorationRightsUrl, requestParams).then(function(response) {
+        var requestUrl = '';
```

Comment:
Yes, saveChangeToBackend is not a great function, but that's probably because it corresponded to a backend URL that did a huge bunch of things. It should be removed and the other two functions added (so yes, I do mean /createhandler/rights/ and /createhandler/publish/ -- though the last one should perhaps be /createhandler/status/ since you could unpublish too).

And yes, I do mean breaking up requestParams in the frontend.